adafruit_ads1x15/ads1x15.py

```diff
@@ -69,66 +71,73 @@ def __init__(
         i2c: I2C,
         gain: float = 1,
         data_rate: Optional[int] = None,
-        mode: Mode = Mode.SINGLE,
+        mode: int = Mode.SINGLE,
         address: int = _ADS1X15_DEFAULT_ADDRESS,
     ):
         # pylint: disable=too-many-arguments
         self._last_pin_read = None
         self.buf = bytearray(3)
-        self._data_rate = self._gain = self._mode = None
+        self._gain = gain
```

Comment:
You should be able to just use the setters, which set the underlying hidden variables themselves:

```python
self.gain = gain
```

Basically, from the original, the only change needed (unless something like pylint gets angry) is to remove setting everything to `None` prior.  If it does get upset, honestly setting them to `None` isn't bad - they'll all get overwritten in the setters, and given the implementation that nothing will initialize if valid values aren't selected, there isn't a need to type any of the getters as `Optional`.

Reply:
Feel free to request a review after address this so I don't block you :)